src/peer_select.cc

```diff
@@ -572,58 +743,139 @@ PeerSelector::selectPinned()
 void
 PeerSelector::selectSomeNeighbor()
 {
-    CachePeer *p;
-    hier_code code = HIER_NONE;
-    assert(entry->ping_status == PING_NONE);
-
-    if (direct == DIRECT_YES) {
-        entry->ping_status = PING_DONE;
-        return;
-    }
+    assert(entry);
```

Comment:
Why do we need to assert on the entry? I do not see any uses of `entry` here.

Reply:
I can not find the related code.
If you refer to the one inside  selectSomeNeighbor() I move it to PeerSelector::selectMore before call selectSomeNeighbor().
In any case the entry is important. For example to define possible requests which can be cached. It does not make sense to ping for cached requests the peers for example.